components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricHandler.java

```diff
@@ -0,0 +1,440 @@
+                Map<String, String[]> apiMap = new HashMap();
+                apiMap.put(MetricConstants.API_LATENCY_SECONDS, new String[]{serviceName, serviceType,
+                        apiInvocationUrl});
+
+                synCtx.setProperty(MetricConstants.API_LATENCY_TIMER,
+                        metricReporterInstance.getTimer(MetricConstants.API_LATENCY_SECONDS, apiMap));
+                break;
+            default:
+                log.error("No proper service type found");
+                break;
+        }
+    }
+
+    /**
+     * Stop the timer when the response is leaving the Synapse Engine.
+     *
+     * @param timer The Timer object used to observe latency
+     * @ param synCtx The Synapse Message Context
+     */
+    private void stopTimers(Object timer, MessageContext synCtx) {
+        if (null == synCtx.getProperty(SynapseConstants.IS_ERROR_COUNT_ALREADY_PROCESSED) && (null != timer)) {
+             metricReporterInstance.observeTime(timer);
+        }
+    }
+
+    /**
+     * Increment the request count received by a proxy service.
+     *
+     * @param name The metric name
+     */
+    private void incrementProxyCount(String name) {
+        Map<String, String[]> metricCountMap = new HashMap();
+        metricCountMap.put(MetricConstants.PROXY_REQUEST_COUNT_TOTAL, new String[] {name,
+                SynapseConstants.PROXY_SERVICE_TYPE});
+        metricReporterInstance.incrementCount(MetricConstants.PROXY_REQUEST_COUNT_TOTAL, metricCountMap);
+    }
+
+    /**
+     * Increment the request count received by an api.
+     *
+     * @param name             The metric name
+     * @param apiInvocationUrl api Invocation URL
+     */
+    private void incrementAPICount(String name, String apiInvocationUrl) {
+        Map<String, String[]> metricCountMap = new HashMap();
+        metricCountMap.put(MetricConstants.API_REQUEST_COUNT_TOTAL, new String[]{name,
+                SynapseConstants.FAIL_SAFE_MODE_API, apiInvocationUrl});
+        metricReporterInstance.incrementCount(MetricConstants.API_REQUEST_COUNT_TOTAL, metricCountMap);
+    }
+
+    /**
+     * Increment the request count received by an inbound endpoint.
+     *
+     * @param name The metric name
+     */
+    private void incrementInboundEndPointCount(String name) {
+        Map<String, String[]> metricCountMap = new HashMap();
+        metricCountMap.put(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_TOTAL, new String[]{name,
+                MetricConstants.INBOUND_ENDPOINT});
+        metricReporterInstance.incrementCount(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_TOTAL, metricCountMap);
+    }
+
+    /**
+     * Increment the error request count received by a proxy service.
+     *
+     * @param name The metric name
+     */
+    private void incrementProxyErrorCount(String name) {
+        Map<String, String[]> metricCountMap = new HashMap();
+        metricCountMap.put(MetricConstants.PROXY_REQUEST_COUNT_ERROR_TOTAL, new String[]{name,
```

Comment:
Can't we just pass the metric name and set of properties rather than a Map?

Reply:
Changed to pass the set of properties